controller/search_blackbox_test.go

```diff
@@ -63,6 +67,151 @@ func (s *searchControllerTestSuite) SetupTest() {
+		// when
+		filter := fmt.Sprintf(`{"space": "%s"}`, fxt.WorkItems[0].SpaceID)
+		rr := httptest.NewRecorder()
+		goaCtx := goa.NewContext(s.svc.Context, rr, nil, nil)
+		rw := test.WorkitemsCSVSearchOK(s.T(), goaCtx, s.svc, s.controller, &filter, nil, nil, nil)
+		// then
+		recorder := rw.(*httptest.ResponseRecorder)
+		recorder.Flush()
+		require.NotNil(t, recorder.Body)
+		bodyStr := recorder.Body.String()
+		// deserialize and check consistency of header and entity lines.
+		entities, err := deserialize(bodyStr)
+		require.NoError(t, err)
+		require.Len(t, entities, 1)
+		// do some consistency checks - a full serialization check is
+		// done down the line in the simple_type and compound type tests.
+		t.Run("simple type", func(t *testing.T) {
+			require.Equal(t, fxt.WorkItems[0].Fields[workitem.SystemTitle], entities[0]["Title"])
+		})
+		t.Run("compound type", func(t *testing.T) {
+			require.Equal(t, "important;backend", entities[0]["Labels"])
+		})
+		t.Run("resolved type", func(t *testing.T) {
+			require.Equal(t, "New", entities[0]["State"])
+		})
+		t.Run("header format", func(t *testing.T) {
+			require.NotEmpty(t, rw.Header().Get("Content-Disposition"))
+			// extract time string from filename returned in the header
+			r, _ := regexp.Compile("^attachment; filename='workitems-(.*)-[0-9]+.csv'$")
+			timeStr := r.FindStringSubmatch(rw.Header().Get("Content-Disposition"))[1]
+			// parse the time string, make sure it is a real date according to RFC3339
+			parsedTime, err := time.Parse(time.RFC3339, timeStr)
+			require.NoError(t, err)
+			// check if the timestamp is consistent
+			assert.WithinDuration(t, time.Now().UTC(), parsedTime, 10*time.Second)
+		})
+	})
+	s.T().Run("empty result", func(t *testing.T) {
+		newFixture(t, 1)
+		// when giving a non-existing SpaceID
+		filter := fmt.Sprintf(`{"space": "%s"}`, uuid.NewV4().String())
+		rr := httptest.NewRecorder()
+		goaCtx := goa.NewContext(s.svc.Context, rr, nil, nil)
+		rw := test.WorkitemsCSVSearchOK(s.T(), goaCtx, s.svc, s.controller, &filter, nil, nil, nil)
+		// then
+		recorder := rw.(*httptest.ResponseRecorder)
+		recorder.Flush()
+		require.NotNil(t, recorder.Body)
+		bodyStr := recorder.Body.String()
+		// deserialize and check consistency of header and entity lines.
+		entities, err := deserialize(bodyStr)
+		require.NoError(t, err)
+		require.Len(t, entities, 0)
+	})
+	s.T().Run("paging", func(t *testing.T) {
+		fxt := newFixture(t, 10)
+		// when
+		filter := fmt.Sprintf(`{"space": "%s"}`, fxt.WorkItems[0].SpaceID)
+		rr := httptest.NewRecorder()
+		goaCtx := goa.NewContext(s.svc.Context, rr, nil, nil)
+		moreMsg := "\nWIT_NOTE_MORE: There are more result entries. You may want to narrow down your query or use paging to retrieve more results."
+		var entities1 []map[string]string
+		t.Run("window 1", func(t *testing.T) {
+			// fetch window 1
+			rw := test.WorkitemsCSVSearchOK(s.T(), goaCtx, s.svc, s.controller, &filter, nil, ptr.Int(5), ptr.Int(0))
```

Comment:
Don't you have to check for correct mimetype headers as well?

Reply:
Added check. Fixed b401deb.